commons-app/apps-android-commons
# [WIP] Enhancement: Tapping the "nearest place" card view on the main screen in 2.9 should bring user to Nearby map with that particular pin selected (#2034)

**Description (required)**

Fixes #2034 Tapping the "nearest place" card view on the main screen in 2.9 should bring user to Nearby map with that particular pin selected

What changes did you make and why?

I added two functions in order to implement the enhancement.

The first function, selectNearestMarker, will programatically select the first item in the list of nearby places, which is sorted by distance from nearest to furthest.

The second function, deselectAllMarkers, is used to remove the info window of all selected markers. This was implemented after testing, because if you do not de-select the marker and close its info window, you would end up with many 

**Tests performed (required)**

Tested with the 2.9 branch on a Samsung Galaxy S8+ with API level 27.

Test procedure:
1) Open the app and Sign in. The Home page loads your contributions.

2) Click the "nearest place" card. The Nearby fragment housing the map is loaded. The nearest marker is selected, and its associated infowindow is displayed.

3) Move the camera. The map, markers, and infowindow should shift on the screen as the camera moves. The infowindow should stay open and maintain its position relative to its marker.

3) Zoom in and out. The map should zoom out, the marker and infowindow should stay open.

4) Change orientation. The fragment should be reloaded in the new configuration. The markers and infowindow should still all be displayed.

5) click on the selected marker. The marker is de-selected and the infowindow is closed.

6) click on the marker again. The infowindow is re-opened and information about the location is displayed below

7) click on a different marker. The infowindow and information about the new marker are displayed.
